src/documentation/ModelArea.tsx

```diff
@@ -0,0 +1,223 @@
+import Highlight from "./reference/Highlight";
+import CodeEmbed from "./common/CodeEmbed";
+import OpenButton from "../project/OpenButton";
+import { useModelData } from "./ml/use-model-data";
+import { ChangeEvent, useCallback, useState } from "react";
+
+interface ModelTopicEntry {
+  name: string;
+  text: string;
+  code: string | ((actionName: string) => string);
+  slug: string;
+  detail?: string;
+  alternativesLabel?: string;
+}
+
+const modelTopicEntries: ModelTopicEntry[] = [
+  {
+    name: "Class names",
+    text: "Return the class names of the current model.",
+    code: "import model\n\n\nnamesList = model.get_class_names()",
+    slug: "class-names",
+  },
+  {
+    name: "Current action",
+    text: "Return the current recognised action.",
+    code: "import model\n\n\ndisplay.scroll(model.current_action())",
+    slug: "current-action",
+  },
+  {
+    name: "Is action",
+    text: "Check if an action is currently being performed.",
+    code: (actionName) =>
+      `import model\n\n\nwhile True:\n    if model.is_action('${actionName}'):\n        display.scroll('Yes')`,
+    slug: "is-action",
+    detail: "Triggered on the current recognised action.",
+    alternativesLabel: "Select action:",
+  },
+  {
+    name: "Was action",
+    text: "Check if an action was performed.",
+    code: (actionName) =>
+      `import model\n\n\nwhile True:\n    if model.was_action('${actionName}'):\n        display.scroll('Yes')`,
+    slug: "was-action",
+    detail: "Triggered if the specified action was recognised since last time.",
+    alternativesLabel: "Select action:",
+  },
+];
+
+const ModelArea = () => {
+  const [modelData] = useModelData();
+  return (
+    <HeadedScrollablePanel
+      heading={
+        <AreaHeading
+          name="Machine learning"
+          description="Use the machine learning model that was trained"
+        />
+      }
+    >
+      <Box
+        p={5}
+        pb={1}
+        fontSize="sm"
+        sx={{
+          ...docStyles,
+        }}
+      >
+        <Stack spacing={3}>
+          <Text>Open your data.json file to get started.</Text>
+          <OpenButton alignSelf="flex-start" fileType="mljson" mode="button" />
+        </Stack>
+      </Box>
+      {modelData.length > 0 && (
+        <>
+          <Box
+            p={5}
+            pb={1}
+            fontSize="sm"
+            sx={{
+              ...docStyles,
+            }}
+          >
+            <Text>Some optional introduction text</Text>
+          </Box>
+          <List flex="1 1 auto">
+            {modelTopicEntries.map((entry) => (
+              <ListItem key={entry.slug}>
+                <ModelTopicEntry content={entry} />
+                <Divider />
+              </ListItem>
+            ))}
+          </List>
+        </>
+      )}
+    </HeadedScrollablePanel>
+  );
+};
+
+interface MlItemProps {
+  content: ModelTopicEntry;
+}
+
+const ModelTopicEntry = ({ content }: MlItemProps) => {
```

Comment:
This whole component is hacky. If we had a markdown alternative for rendering extension documentation, we'd use it here. It's pretty horrible to write the model topic entries in the Sanity format.